forte/data/ontology/top.py

```diff
@@ -686,5 +690,276 @@ def get(
+        within the grid cell will masked as zeros. The array entries that are
+        within the grid cell will be copied to the zeros numpy array.
+
+
+        Args:
+            h_idx: the zero-based index of the grid cell of the first
+                dimension.
+            w_idx: the zero-based index of the grid cell of the second
+                dimension.
+
+        Raises:
+            ValueError: ``h_idx`` is out of the range specified by ``height``.
+            ValueError: ``w_idx`` is out of the range specified by ``width``.
+
+        Returns:
+            numpy array that represents the grid cell.
+        """
+        if not 0 <= h_idx < self._height:
+            raise ValueError(
+                f"input parameter h_idx ({h_idx}) is"
+                "out of scope of h_idx range"
+                f" {(0, self._height)}"
+            )
+        if not 0 <= w_idx < self._width:
+            raise ValueError(
+                f"input parameter w_idx ({w_idx}) is"
+                "out of scope of w_idx range"
+                f" {(0, self._width)}"
+            )
+
+        # initialize a numpy zeros array
+        array = np.zeros(self.img_arr.shape)
+        # set grid cell entry values to the values of the original image array
+        # (entry values outside of grid cell remain zeros)
+        # An example of computing grid height index range is
+        # index * cell height : (index + 1) * cell height.
+        # It's similar for computing cell width index range
+        array[
+            h_idx * self.c_h : (h_idx + 1) * self.c_h,
+            w_idx * self.c_w : (w_idx + 1) * self.c_w,
+        ] = self.img_arr[
+            h_idx * self.c_h : (h_idx + 1) * self.c_h,
+            w_idx * self.c_w : (w_idx + 1) * self.c_w,
+        ]
+        return array
+
+    def get_grid_cell_center(self, h_idx: int, w_idx: int):
+        return (
+            (h_idx * self.c_h + (h_idx + 1) * self.c_h) // 2,
+            (w_idx * self.c_w + (w_idx + 1) * self.c_w) // 2,
+        )
+
+    @property
+    def image_payload_idx(self) -> int:
+        return self._image_payload_idx
+
+    @property
+    def num_grid_cells(self):
+        return self._height * self._width
+
+    @property
+    def height(self):
+        return self._height
+
+    @property
+    def width(self):
+        return self._width
+
+    def __eq__(self, other):
+        if other is None:
+            return False
+        return (self.image_payload_idx, self._height, self._width) == (
+            other.image_payload_idx,
+            self._height,
+            self._width,
+        )
+
+
+class Region(ImageAnnotation):
+    def __init__(self, pack: PackType, image_payload_idx: int):
+        super().__init__(pack, image_payload_idx)
+        if image_payload_idx is None:
+            self._image_payload_idx = 0
+        else:
+            self._image_payload_idx = image_payload_idx
+        self.img_arr = self.pack.get_image_array(self._image_payload_idx)
```

Comment:
I think we don't need to store this variable:

1. if it is just a pointer, it is trivial to use anyway, you can create a property function instead of a variable
2. if it is actual memory copy, it won't be necessary